src/cli/tls_http_server.cpp

```diff
@@ -393,79 +145,147 @@ class TLS_Asio_HTTP_Session final : public std::enable_shared_from_this<TLS_Asio
-            m_status(max_clients) {
-         serve_one_session();
+std::string summarize_request(const Botan::TLS::Stream<tcp_stream>& tls_stream,
+                              const http::request<http::string_body>& req) {
+   std::ostringstream strm;
+
+   const auto& remote = tls_stream.next_layer().socket().remote_endpoint();
+
+   strm << "Client " << remote.address().to_string() << " requested " << req.method_string() << " " << req.target()
+        << "\n";
+
+   if(std::distance(req.begin(), req.end()) > 0) {
+      strm << "Client HTTP headers:\n";
+      for(const auto& header : req) {
+         strm << " " << header.name_string() << ": " << header.value() << "\n";
       }
+   }
 
-   private:
-      void serve_one_session() {
-         auto new_session = session::create(get_io_service(m_acceptor), m_session_manager, m_creds, m_policy);
+   return strm.str();
+}
 
-         m_acceptor.async_accept(
-            new_session->client_socket(),
-            boost::bind(&TLS_Asio_HTTP_Server::handle_accept, this, new_session, boost::asio::placeholders::error));
+auto make_final_completion_handler(const std::shared_ptr<Logger>& logger, const std::string& context) {
+   return [=](std::exception_ptr e) {
+      if(e) {
+         try {
+            std::rethrow_exception(std::move(e));
+         } catch(const std::exception& ex) {
+            logger->error(Botan::fmt("{}: {}", context, ex.what()));
+         }
       }
+   };
+}
 
-      void handle_accept(const session::pointer& new_session, const boost::system::error_code& error) {
-         if(!error) {
-            new_session->start();
-            m_status.client_serviced();
+std::shared_ptr<http::response<http::string_body>> handle_response(const http::request<http::string_body>& req,
+                                                                   const std::shared_ptr<TlsHttpCallbacks>& callbacks,
+                                                                   const Botan::TLS::Stream<tcp_stream>& tls_stream,
+                                                                   const std::shared_ptr<Logger>& logger) {
+   logger->log(Botan::fmt("{} {}", req.method_string(), req.target()));
+
+   auto response = [&] {
+      if(req.method() != http::verb::get) {
+         auto resp =
+            std::make_shared<http::response<http::string_body>>(http::status::method_not_allowed, req.version());
+         resp->body() = std::string("Unsupported HTTP verb\n");
+         return resp;
+      } else if(req.target() == "/" || req.target() == "/status") {
+         auto resp = std::make_shared<http::response<http::string_body>>(http::status::ok, req.version());
+         resp->body() = callbacks->summary() + summarize_request(tls_stream, req);
+         return resp;
+      } else {
+         auto resp = std::make_shared<http::response<http::string_body>>(http::status::not_found, req.version());
+         resp->body() = std::string("Not found");
+         return resp;
+      }
```

Comment:
I guess it's not a huge deal but this lambda seems really odd to me. There are really only two things that change depending on the params: the response code and the body. Instead there is a lot of repetition in the three cases. Could we do something like

```
// 
auto [response_code, body] = ...;
auto resp = std::make_shared<...>(response_code, req.version());
resp->body() = std::move(body);
```